### PR TITLE
CustomReport | CSV Download with progress bar #6408

### LINKED ISSUE #6408: CustomReport | CSV Download | Add pagination
# Bug Report

## Problem Description
Currently on CSV-Download, the entire data is fetched at once from the database. 
Offset-/Limit is not taken into consideration internally:

This causes internal server errors / SQL timeouts, if many datasets are part of the result (e.g. 600.000 records). 

## Solution Suggestion
Fetch the dataset using blocks shouldn't be that hard.
Line:
https://github.com/pimcore/pimcore/blob/master/bundles/AdminBundle/Controller/Reports/CustomReportController.php#L409

One just has to verify if the SQL statement defined in the custom report already defines offset / limit on its own. If yes, then the download can remain the same. If no, then pagination can be applied.

We can provide a PR if you like the suggestion.

### PR DESCRIPTION
**Pull request by PGBrandi**
## Changes in this pull request  
Resolves #6408
The creation of the file gets split in several chunks to prevent a timeout. The progress is returned after each chunk and the client keeps resuming until the server says that it's finished with the creation of the file.
A progress bar to show the current progress was also added.

### CODE REVIEW

**Comment by brusch** [comment]
@PGBrandi Please fix the tests / PhpStan, see: 
https://travis-ci.com/github/pimcore/pimcore/jobs/342543011#L1189

Thanks a lot!

File: bundles/AdminBundle/Controller/Reports/CustomReportController.php [hunk 1]
```diff
@@ -369,14 +370,13 @@ public function chartAction(Request $request)
     }
 
     /**
-     * @Route("/download-csv", name="pimcore_admin_reports_customreport_downloadcsv", methods={"GET"})
+     * @Route("/create-csv", name="pimcore_admin_reports_customreport_createcsv", methods={"GET"})
      *
      * @param Request $request
      *
      * @return BinaryFileResponse
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Reports/CustomReportController.php:377]:
```suggestion
     * @return JsonResponse
```

**Comment by weisswurstkanone** [comment]
What will be the outcome if the data changes in the meantime?

**Comment by weisswurstkanone** [comment]
@PGBrandi ping

**Comment by PGBrandi** [comment]
> What will be the outcome if the data changes in the meantime?

It depends on the changed data set. If one entry was removed/changed after already being processed, the data simply stays in the file the way it was. However, the new data from unprocessed entries is updated in the file.

**Pull request merged**